### ISSUE TITLE
Minelayer Vehicles Cannot Continue Building a Mine if They Stop the Mine's Construction

### ISSUE THREAD

**Issue by Brian-Catcow-B** [beyond-all-reason/Beyond-All-Reason#3210]
### Description

If a Mine is not fully built and nothing is building it, a minelayer cannot pick up the building again

### Expected Behaviour

Minelayer vehicles should be able to assist in constructing any of the buildings available to it, so that if a command is issued, cancelling the progress of a mine being built, then another command can be issued to continue the progress of the mine

### Actual Behaviour

the only way for a minelayer vehicle to construct anything is for it to start the construction of it

### Reproduction steps

start building a heavy mine (works with anything, really) with a minelayer vehicle. Then cancel construction. Then realize that you cannot issue a command to continue construction on heavy mine.

**Comment by hihoman23** [comment]
I don't think that this is a bug. This is done this way so that minelayers can't build assist each other.

**Comment by efrec** [comment]
Admittedly a defect but very minor in consequence. Like hihoman23 said, this is working as intended, so not a bug. Other contructors have no problem assisting a minelayer, but minelayers cannot assist.

An enhancement could allow minelayers to continue their own solo builds that were interrupted. Otherwise, they must not be able to assist any worker task. This would need a gadget using `AllowCommand` and `RegisterAllowCommand` and that can recognize both minelayers and mines. I'm not sure this change is wanted or needed, though.

**Comment by WatchTheFort** [comment]
If minelayers _could_ assist construction, would that be so bad?

**Comment by efrec** [comment]
I think this is just some OTAism that minelayers build *only* mines.

**Comment by drivver44** [comment]
> If minelayers _could_ assist construction, would that be so bad?

no
mine layers have very little bp that it wouldnt matter.
its less reason to not build them.
they still are bad to build outside of minelaying

edit
armmlv is at 40 bp and armveh con is at 90
cormlv is at 40 bp and corveh con is at 95 bp
